sql3/planner/inbuiltfunctionsstring.go

```diff
@@ -644,3 +668,65 @@ func (n *callPlanExpression) EvaluateReplicate(currentRow []interface{}) (interf
 	return strings.Repeat(stringArg, int(intArg)), nil
 
 }
+
+// function to get the index of substring in the given string
+func (n *callPlanExpression) EvaluateCharIndex(currentRow []interface{}) (interface{}, error) {
+
+	argEval, err := n.args[0].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+	if argEval == nil {
+		return nil, nil
+	}
+	subString, ok := argEval.(string)
+	if !ok {
+		return nil, sql3.NewErrInternalf("unexpected type converion %T", argEval)
+	}
+
+	argEval, err = n.args[1].Evaluate(currentRow)
+	if err != nil {
+		return nil, err
+	}
+	if argEval == nil {
+		return nil, nil
+	}
+	inputString, ok := argEval.(string)
+	if !ok {
+		return nil, sql3.NewErrInternalf("unexpected type converion %T", argEval)
+	}
+
+	if len(n.args) == 2 {
+		res := strings.Index(inputString, subString)
+		if res > -1 {
```

Comment:
do we need this if condition? 
// return int64(res), nil

Reply:
yes, when position is not found then it has to return -1

Reply:
The strings.Index() returns -1 if not found, which is what we are returning in line 705.